src/abe_policy/parser.rs

```diff
@@ -0,0 +1,267 @@
+    ///
+    /// The policy specifications must be passed as a mapping object:
+    /// ```json
+    ///     {
+    ///        "Security Level::<": [
+    ///            "Protected",
+    ///            "Confidential",
+    ///            "Top Secret::+"
+    ///        ],
+    ///        "Department": [
+    ///            "R&D",
+    ///            "HR",
+    ///            "MKG",
+    ///            "FIN"
+    ///        ]
+    ///    }
+    /// ```
+    fn try_from(value: HashMap<String, Vec<String>>) -> Result<Self, Self::Error> {
+        let mut policy = Self::new();
+
+        for (axis, attributes) in &value {
+            // Split the axis into axis name and hierarchy flag
+            let (axis_name, hierarchical) = match axis.split_once("::") {
+                Some((name, specs)) => {
+                    // If the axis contains the hierarchy flag, parse it
+                    let hierarchical = match specs {
+                        "<" => true,
+                        x => {
+                            return Err(Error::ConversionFailed(format!("unknown axis spec {x}")));
+                        }
+                    };
+                    (name, hierarchical)
+                }
+                // If there is no hierarchy flag, assume the axis is non-hierarchical
+                None => (axis.as_str(), false),
+            };
+
+            let mut attributes_properties: Vec<(&str, EncryptionHint)> =
+                Vec::with_capacity(attributes.len());
+
+            // Parse each attribute and its encryption hint
+            for att in attributes {
+                let (att_name, encryption_hint) = match att.split_once("::") {
+                    Some((name, specs)) => {
+                        let encryption_hint = match specs {
+                            "+" => EncryptionHint::Hybridized,
+                            x => {
+                                return Err(Error::ConversionFailed(format!(
+                                    "unknown attribute spec {x}"
+                                )));
+                            }
+                        };
+                        (name, encryption_hint)
+                    }
+                    // If there is no encryption hint, assume the attribute is non-hybridized
+                    None => (att.as_str(), EncryptionHint::Classic),
+                };
+                attributes_properties.push((att_name, encryption_hint));
+            }
+
+            // Add the axis to the policy
+            policy.add_dimension(DimensionBuilder::new(
+                axis_name,
+                attributes_properties,
+                hierarchical,
+            ))?;
+        }
+        Ok(policy)
+    }
+}
+
+fn convert_attribute(attribute: (String, AttributeParameters)) -> String {
+    let (name, params) = attribute;
+    match params.get_encryption_hint() {
+        EncryptionHint::Hybridized => name + "::+",
+        EncryptionHint::Classic => name,
+    }
+}
```

Comment:
Maybe just declare this function as a local variable. It does not make sens outside of the scope of the `try_from` context.